src/main/scala/api/other/RegistrationComparison.scala

```diff
@@ -26,4 +27,25 @@ object RegistrationComparison {
     val hausdorffDistance = MeshMetrics.hausdorffDistance(reconstruction, groundTruth)
     println(s"ID: ${id} average2surface: ${avgDist2Surf} hausdorff: ${hausdorffDistance}")
   }
+
+  private def avgDistanceBoundaryAware(m1: TriangleMesh[_3D], m2: TriangleMesh[_3D]): (Double, Double) = {
+
+    val pointsOnSample = m1.pointSet.points
+    val dists = for (p <- pointsOnSample) yield {
+      val pTarget = m2.operations.closestPointOnSurface(p).point
```

Comment:
the inside of the loop could be replaced with the following one (the for loop could yield an option and then there is not the need to filter, but just a flatten.... ):
```scala
  val instance: TriangleMesh3D = ???
  val target: TriangleMesh3D = ???

  for ( instancePoint <- instance.pointSet.points ) yield {
    val closestTargetPoint: ClosestPointWithType = target.operations.closestPointOnSurface(instancePoint)
    closestTargetPoint match {
      case ClosestPointIsVertex(_, distanceSquared, pid) =>
        if (target.operations.pointIsOnBoundary(pid)) -1.0
        else Math.sqrt(distanceSquared)
      case ClosestPointOnLine(point, distanceSquared, pids, _) =>
        if (target.operations.edgeIsOnBoundary(pids._1,pids._2)) -1.0
        else Math.sqrt(distanceSquared)
      case ClosestPointInTriangle(_, distanceSquared, _, _) =>
        Math.sqrt(distanceSquared)
    }
  }
```